Sources/Datadog/Core/DD/Telemetry.swift

```diff
@@ -92,3 +92,8 @@ extension Telemetry {
         self.error(message, error: DDError(error: error), file: file, line: line)
     }
 }
+
+internal struct NoOpTelemetry: Telemetry {
```

Comment:
Sounds good, changed 👍. 

It would be also great to make it consistent among the entire codebase + enforced by the linter, so I created JIRA: _`RUMM-2340` Enforce consistent `NOP` prefix for all no-op implementations_